test_common/harness/testHarness.h

```diff
@@ -150,17 +150,22 @@ template <typename T> T *register_test(const char *name, Version version)
 
 #define REGISTER_TEST(name) REGISTER_TEST_VERSION(name, Version(1, 2))
 
-#define REQUIRE_EXTENSION(name)                                                \
+#define REQUIRE_EXTENSION(ext)                                                 \
     do                                                                         \
     {                                                                          \
-        if (!is_extension_available(device, name))                             \
+        if (!is_extension_available(device, ext##_EXTENSION_NAME,              \
+                                    ext##_EXTENSION_VERSION))                  \
```

Comment:
This may be OK, but so we go in with eyes open - I think this means that the tests (that use this macro at least) will only run for implementations that support  the extension version in the headers, and only that version.

Do we want to have the ability to test other extension versions also?  For example, I could envision tests for an extension that have one codepath for version 1.X and another codepath for version 2.X, kind of like the tests for integer dot product, but perhaps with a breaking change between 1.x and 2.X.  For usages like this one I don't think we'd be able to use this REQUIRE_EXTENSION macro the way it is currently defined.